docs/src/extend/custom-rules-tutorial.md

```diff
@@ -0,0 +1,200 @@
+
+```js
+// foo-bar.js
+
+module.exports = {
+    meta: {
+        // Specify that provides a suggested change in output
+        type: "suggestion",
+        // Description of rule. Useful for tools that generate
+        // documentation from the rule source.
+        docs: {
+            description: "Can only assign 'bar' to `const foo`.",
+        },
+        fixable: "code"
+    },
+    create(context) {
+        return {
+           // TODO: add callback function(s)
+        };
+    }
+};
+```
+
+To learn more about rule metadata, refer to [Rule Structure](custom-rules#rule-structure).
+
+## Step 4: Add Rule Callback Functions
+
+TODO: describe
+
+```js
+// foo-bar.js
+
+module.exports = {
+    meta: {
+        type: "fix",
+        docs: {
+            description: "Can only assign 'bar' to `const foo`.",
+        },
+         fixable: "code"
+    },
+    create(context) {
+        return {
+            // Performs action in the function on every variable declaration
+            "VariableDeclaration": function(node) {
+                // Check if a `const` variable declaration
+                if(node.kind === "const") {
+                    // Check if variable name is `foo`
+                    if(node.declarations[0].id.name === "foo") {
+                        // Check if value of variable is "bar"
+                        if (node.declaration.init.value !== "bar") {
+                            // Report error to ESLint. Error message uses
+                            // a message placeholder to include the incorrect value
+                            // in the error message.
+                            // Also includes a `fix(fixer)` function that replaces
+                            // any values assigned to `const foo` with "bar".
+                            context.report({
+                                node,
+                                message: 'Value other than "bar" assigned to \`const foo\`. Unexpected value: {{ notBar }}',
+                                data: {
+                                    notBar: node.declaration.init.value
+                                },
+                                fix(fixer) {
+                                    return fixer.replaceText(node.declaration.init, "bar");
+                                }
+                            });
+                        }
+                    }
+                }
+            }
+        };
+    }
+};
+```
+
+## Step 5: Set up Testing
+
+With the rule written, you can test it to make sure it's working as expected. First you need to set up the testing environment in your project.
+
+This tutorial will use the popular JavaScript testing library [Jest](https://jestjs.io/) and the [ESLint Node.js API](../integrate/nodejs-api) to test out the `foo-bar` rule's functionality.
+
+Install the `jest` and `eslint` package as development dependencies to use them in your tests:
```

Comment:
```suggestion
Install the `jest` and `eslint` packages as development dependencies to use them in your tests:
```

Reply:
```suggestion
Install `jest` and `eslint` package as development dependencies to use them in your tests:
```
I feel like this sounds better?

Reply:
sorry for confusion this file is a previous version of the tutorial, and this content isn't in the current version.